charts/shoot-core/components/charts/network-policies/templates/allow-to-dns.yaml

```diff
@@ -20,10 +20,8 @@ spec:
     - podSelector:
         matchExpressions:
         - {key: k8s-app, operator: In, values: [kube-dns]}
-    {{- if .Values.nodeLocalDNS.enabled }}
     - ipBlock:
         cidr: {{ .Values.nodeLocalDNS.kubeDNSClusterIP }}/32
-    {{- end }}
```

Comment:
In the shoot, why is this removed? The network policy is only required when the shoot enables the node-local-dns, isn't it?

Reply:
As discussed with you and @DockToFuture , I adapted all three network policies (garden namespace in seed, control plane(s) in seed, kube-system in shoot) to always allow the node local dns entries, i.e. remove the if checks.
You can argue though that while it is required to remove them from the network policies in the garden namespace and the control plane(s) unless you want to live with cascading effects, it is possible to let this if block stay in the kube-system network policy as this should not affect anything apart from the cluster itself.
I think removing the node local checks on all three network policies is more consistent, but this is not a hard opinion. So if you insist, I can re-add the if check here.

Reply:
To my understanding, we only discussed about always allowing both CIDRs for the network policies created by gardenlet's seed controller due to the fact that we don't know which DNS component (coredns/node-local-dns) is running in the seed. For the shoot, however, we definitely know it because it's configured in the `Shoot` resource. Hence, I would indeed only allow required network traffic if we have the possibility to shrink it down.

Reply:
Ok, I reverted the changes related to the network policy in the kube-system namespace.
As a positive side effect, it reduces the amount of files being changed.